src/main/kotlin/resource/PokemonResource.kt

```diff
@@ -0,0 +1,34 @@
+package resource
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.ktor.application.call
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.route
+import service.pokemon.PokemonService
+import utility.PokeApi
+import utility.PokeApiAdapter
+
+fun Route.pokemon(pokemonService: PokemonService) {
+    route("/pokemon") {
+        get("/{id}") {
+            val id = call.parameters["id"]
+            if (id != null) {
```

Comment:
Not needed

Reply:
Isn't it? How would you implement it?

Reply:
It turned out that `ktor` will not even call this endpoint handler when the query does not match.
Meaning that if no `id` is supplied, the handler will never be called and a `404` is directly returned instead as no handler matched the request. That's why we don't need to do this nullability check there.

You could not know though of course :)